### PR TITLE
Polish build

### PR DESCRIPTION
**Pull request by Goooler**
- Migrate to Gradle.
- Update submodule.
- Result: https://github.com/Goooler/dalvik-dx/actions/runs/5978447249

### CODE REVIEW

**Comment by JakeWharton** [comment]
I will get to this...

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+org.gradle.caching=true
+org.gradle.jvmargs=-Dfile.encoding=UTF-8
+org.gradle.parallel=true
```
Comment (JakeWharton) [inline gradle.properties]:
```suggestion
org.gradle.parallel=true

```

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+plugins {
```
Comment (JakeWharton) [inline build.gradle.kts]: Please use Groovy and tabs

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Build
```
Comment (JakeWharton) [inline .github/workflows/build.yml:1]: Add Gradle validation action as separate workflow like other repos, please.

**Review by JakeWharton — COMMENTED** [review]
Starting to catch up on some of these. Sorry for delay.

File: build.gradle.kts (hunk 2 above)
Comment (JakeWharton) [reply build.gradle.kts]: And trailing newline!

Comment (Goooler) [reply build.gradle.kts]: a60365616650e6f8b4b1d17c5b624348ee83406e

File: .github/workflows/build.yml (hunk 3 above)
Comment (Goooler) [reply .github/workflows/build.yml:1]: 90f95b9a53461907f47b3e8c9e83713cb66e3058

**Comment by Goooler** [comment]
```sh
tree ~/.m2/repository

~/.m2/repository
└── com
    └── jakewharton
        └── android
            └── repackaged
                └── dalvik-dx
                    ├── 14.0.0_r18-SNAPSHOT
                    │   ├── dalvik-dx-14.0.0_r18-SNAPSHOT-javadoc.jar
                    │   ├── dalvik-dx-14.0.0_r18-SNAPSHOT-sources.jar
                    │   ├── dalvik-dx-14.0.0_r18-SNAPSHOT.jar
                    │   ├── dalvik-dx-14.0.0_r18-SNAPSHOT.module
                    │   ├── dalvik-dx-14.0.0_r18-SNAPSHOT.pom
                    │   └── maven-metadata-local.xml
                    └── maven-metadata-local.xml

7 directories, 7 files
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+      - uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: true
+      - name: Publish Artifacts
+        run: ./gradlew publish
+        if: github.ref == 'refs/heads/trunk'
```
Comment (JakeWharton) [inline .github/workflows/build.yml]:
```suggestion
      - run: ./gradlew publish
        if: ${{ github.ref == 'refs/heads/trunk' && github.repository == 'JakeWharton/dalvik-dx' }}
```
No need to build twice

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**